include/crocoddyl/multibody/residuals/pair-collisions.hpp

```diff
@@ -0,0 +1,157 @@
+ * of the residual function are computed analytically.
+ *
+ * As described in `ResidualModelAbstractTpl()`, the residual value and its Jacobians are calculated by `calc` and
+ * `calcDiff`, respectively.
+ *
+ * \sa `ResidualModelAbstractTpl`, `calc()`, `calcDiff()`, `createData()`
+ */
+template <typename _Scalar>
+class ResidualModelPairCollisionsTpl : public ResidualModelAbstractTpl<_Scalar> {
+ public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  typedef _Scalar Scalar;
+  typedef MathBaseTpl<Scalar> MathBase;
+  typedef ResidualModelAbstractTpl<Scalar> Base;
+  typedef ResidualDataPairCollisionsTpl<Scalar> Data;
+  typedef ResidualDataAbstractTpl<Scalar> ResidualDataAbstract;
+  typedef StateMultibodyTpl<Scalar> StateMultibody;
+  typedef DataCollectorAbstractTpl<Scalar> DataCollectorAbstract;
+  typedef pinocchio::GeometryModel GeometryModel;
+  
+  typedef typename MathBase::Vector3s Vector3s;
+  typedef typename MathBase::VectorXs VectorXs;
+  typedef typename MathBase::MatrixXs MatrixXs;
+
+  /**
+   * @brief Initialize the pair collisions residual model
+   *
+   * @param[in] state  State of the multibody system
+   * @param[in] nu   Dimension of the control vector
+   * @param[in] geom_model   Pinocchio geometry model containing the collision pair
+   * @param[in] pair_id   Index of the collision pair in the geometry model
+   * @param[in] joint_id   Index of the nearest joint on which the collision link is attached
+   */
+  
+  ResidualModelPairCollisionsTpl(boost::shared_ptr<StateMultibody> state,
+                             const std::size_t& nu,
+                             boost::shared_ptr<GeometryModel> geom_model,
+                             const pinocchio::PairIndex& pair_id, 
+                             const pinocchio::JointIndex& joint_id);
+
+ 
+  virtual ~ResidualModelPairCollisionsTpl();
+
+   /**
+   * @brief Compute the pair collisions residual
+   *
+   * @param[in] data  Pair collisions residual data
+   * @param[in] x     State point \f$\mathbf{x}\in\mathbb{R}^{ndx}\f$
+   * @param[in] u     Control input \f$\mathbf{u}\in\mathbb{R}^{nu}\f$
+   */
+  virtual void calc(const boost::shared_ptr<ResidualDataAbstract> &data, const Eigen::Ref<const VectorXs> &x,
+                    const Eigen::Ref<const VectorXs> &u);
+
+  /**
+   * @brief Compute the derivatives of the pair collisions residual
+   *
+   * @param[in] data  Pair collisions residual data
+   * @param[in] x     State point \f$\mathbf{x}\in\mathbb{R}^{ndx}\f$
+   * @param[in] u     Control input \f$\mathbf{u}\in\mathbb{R}^{nu}\f$
+   */
+  virtual void calcDiff(const boost::shared_ptr<ResidualDataAbstract> &data, const Eigen::Ref<const VectorXs> &x,
+                        const Eigen::Ref<const VectorXs> &u);
+  
+  virtual boost::shared_ptr<ResidualDataAbstract> createData(DataCollectorAbstract *const data);
+  
+  /**
+   * @brief Return the Pinocchio geometry model
+   */
+  const pinocchio::GeometryModel& get_geometryModel() const;
+
```

Comment:
1. Create a method to get and set the `pair_id_` and `joint_id`. Call them `get_pair_id` and `get_joint_id`
2. Define Python bindings of 1.